### PR TITLE
new and updated PRUNERS tools

### PR DESCRIPTION
**Pull request by lee218llnl**
tools for PRUNERS (Providing Reproducibility for Uncovering Non-deterministic Errors in Runs on Supercomputers). More info at https://github.com/PRUNERS.

### CODE REVIEW

File: var/spack/repos/builtin/packages/archer/package.py [hunk 1]
```diff
@@ -30,12 +30,12 @@ class Archer(Package):
     """ARCHER, a data race detection tool for large OpenMP applications."""
 
     homepage = "https://github.com/PRUNERS/ARCHER"
+    url='https://github.com/PRUNERS/archer/archive/v1.0.0.tar.gz'
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/archer/package.py]: @alalazo @tgamblin How did this line pass the flake8 tests? Shouldn't it be required to have spaces around the '='?

File: var/spack/repos/builtin/packages/pruners-ninja/package.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PrunersNinja(AutotoolsPackage):
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/pruners-ninja/package.py]: Can you delete this empty line?

**Comment by lee218llnl** [comment]
@adamjstewart Thanks for looking this over. I pushed your suggested changes. I hope you don't mind that I squashed them too. I will conjecture that the no spaces around "="  may have passed b/c of the exception around lines with "url". I don't know if the exception is specifically for the line length or if it was set to just ignore all style issues.

**Comment by adamjstewart** [comment]
@lee218llnl That's what I thought too, but the exception _is_ specifically for line length:
https://github.com/LLNL/spack/blob/develop/lib/spack/spack/cmd/flake8.py#L55
https://github.com/LLNL/spack/blob/develop/lib/spack/spack/cmd/flake8.py#L71

I wonder if it's because the line is ignored twice? It would end in:
```
# NOQA: ignore=501  # NOQA: ignore=501
```
I'm not sure how that behaves...

**Comment by lee218llnl** [comment]
It looks like the syntax for ignoring flake 8 errors is wrong. Here's a simple test I ran:

[lee218@rzoz-login1:spack]$ cat var/spack/repos/builtin/packages/archer/package.py | grep url
    url      = "https://github.com/PRUNERS/archer/archive/v1.0.0.tar.gz"
    url="https://github.com/PRUNERS/archer/archive/v1.0.0.tar.gz"
    url="https://github.com/PRUNERS/archer/archive/v1.0.0.tar.gz" # NOQA: E501
    url="https://github.com/PRUNERS/archer/archive/v1.0.0.tar.gz" # NOQA: ignore=501
[lee218@rzoz-login1:spack]$ /usr/apps/python/bin/flake8 var/spack/repos/builtin/packages/archer/package.py
var/spack/repos/builtin/packages/archer/package.py:34:8: E225 missing whitespace around operator
var/spack/repos/builtin/packages/archer/package.py:35:8: E225 missing whitespace around operator
var/spack/repos/builtin/packages/archer/package.py:35:66: E261 at least two spaces before inline comment

I will put in a PR to fix this.

**Comment by tgamblin** [comment]
@dongahn: FYI

**Comment by dongahn** [comment]
Very cool! Thanks @tgamblin and @lee218llnl ! 

Spack makes our life a lot easier.

**Pull request merged**